mm2src/coins/utxo.rs

```diff
@@ -2228,3 +782,104 @@ fn kmd_interest(height: Option<u64>, value: u64, lock_time: u64, current_time: u
     minutes -= 59;
     (value / 10512000) * minutes
 }
+
+/// Denominate BigDecimal amount of coin units to satoshis
+pub fn sat_from_big_decimal(amount: &BigDecimal, decimals: u8) -> Result<u64, String> {
+    (amount * BigDecimal::from(10u64.pow(decimals as u32))).to_u64().ok_or(ERRL!("Could not get sat from amount {} with decimals {}", amount, decimals))
+}
+
+pub(crate) fn sign_tx(
+    unsigned: TransactionInputSigner,
+    key_pair: &KeyPair,
+    prev_script: Script,
+    signature_version: SignatureVersion,
+    fork_id: u32,
+) -> Result<UtxoTx, String> {
+    let mut signed_inputs = vec![];
+    for (i, _) in unsigned.inputs.iter().enumerate() {
+        signed_inputs.push(
+            try_s!(p2pkh_spend(&unsigned, i, key_pair, &prev_script, signature_version, fork_id))
+        );
+    }
+    Ok(UtxoTx {
+        inputs: signed_inputs,
+        n_time: unsigned.n_time,
+        outputs: unsigned.outputs.clone(),
+        version: unsigned.version,
+        overwintered: unsigned.overwintered,
+        lock_time: unsigned.lock_time,
+        expiry_height: unsigned.expiry_height,
+        join_splits: vec![],
+        shielded_spends: vec![],
+        shielded_outputs: vec![],
+        value_balance: 0,
+        version_group_id: unsigned.version_group_id,
+        binding_sig: H512::default(),
+        join_split_sig: H512::default(),
+        join_split_pubkey: H256::default(),
+        zcash: unsigned.zcash,
+        str_d_zeel: unsigned.str_d_zeel,
+    })
+}
+
+async fn send_outputs_from_my_address_impl<T>(coin: T, outputs: Vec<TransactionOutput>)
+                                              -> Result<UtxoTx, String>
+    where T: UtxoArcGetter + UtxoArcCommonOps {
+    let _utxo_lock = UTXO_LOCK.lock().await;
+    // TODO replace list_unspent_ordered with ordered_mature_unspents
```

Comment:
TODO?

Reply:
The ordered_mature_unspents method takes MmArc to get path to the database directory, because of this we should add the MmArc to all of this method callers: send_taker_fee, send_maker_payment, send_taker_payment. I think it can be done in the next step - QRC20 swap integration. Or we can do not replace list_unspent_ordered with ordered_mature_unspents.

Reply:
As we will move coins crate to separate lib some day I would prefer for coins implementation to not directly depend on `MmCtx/MmArc` global state. Can we replace MmArc with something like `cache_directory` field in coin struct? 
It will be set up upon coin initialization using DB dir from MmArc in AtomicDEX-API, but will be also possible to be set to something different. Might be also useful for tests because we won't need to build up entire MmArc for coins specific tests.